src/Common/tests/Cryptography/AlgorithmImplementations/RSA/RSAFactory.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Security.Cryptography.Rsa.Tests
+{
+    public interface IRSAProvider
+    {
+        RSA Create();
+        RSA Create(int keySize);
```

Comment:
Thanks; wasn't part of the PR message so I didn't see it (on my phone).  Why does it need to be part of the contract to use it in the tests for that library, which should be using a P2P reference? Presumably we'll add other tests to exercise RSA.Create? Rather than using this abstraction, did we consider using theories and member data to have each test run using all appropriate providers for the current platform, e.g. a function that yields the one provider on Unix and both providers on Windows?